### PR TITLE
Habit details and edits

### LINKED ISSUE #3: US 01.01.01
As a doer, I want to add a habit, giving it a brief title, reason, date to start, and whether it is public or private.

SP: 2
Risk: Low

This User Story is related to the following subtasks:

- [x] #4 
- [x] #5 
- [x] #6 
- [x] #155 
- [x] #77
- [x] #148

### LINKED ISSUE #160: Improve UI for habit editing
This issue is related to user story #12 

Currently, habit editing is occurring through a fragment where the previous values are not even pre-populated (it looks exactly the same as adding a habit). This was involves getting rid of the fragment altogether and putting an activity in its place, with pre-populated values.

### PR DESCRIPTION
**Pull request by PaoVal**
Hello All. This PR will resolve #160 

#160, mostly includes UI modifications to the habit list, habit detailed view list, habit event edit view. 
I provided some much needed user feedback, improved the UI/UX for the detail and edit view. added colour, reduced the amount of text, added icons instead, etc...

This PR resolves #3, #12, #25

#3 - is the user story regarding adding habits. The add habit UI was slightly modified and improved. 
#12 - is the user story regarding habit editing. Amongst many other improvements, the fields are now pre-filled.
#25 - is the user story regarding the habit list. The habit list was edited and finalized with this PR no further changes should occur.

@NickSNFK I think your progress stuff should still be working but I'd appreciate if you could take a look to give it the final go-ahead.

### CODE REVIEW

**Comment by PaoVal** [comment]
(Please ignore my commits where I cited issue 120. That was a mistake and I meant to say #160

File: code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitEditActivity.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import com.example.habittracker.R;
+import com.example.habittracker.utils.CustomDatePicker;
+
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Locale;
+
```
Comment (NickSNFK) [inline code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitEditActivity.java:27]: Javadoc Police.

**Review by NickSNFK — CHANGES_REQUESTED** [review]
Everything looks good to me. Tested and everything works as intended.
See my comment on the JavaDoc for file: HabitEditActivity

File: code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitEditActivity.java (hunk 1 above)
Comment (PaoVal) [reply code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitEditActivity.java:27]: Thanks!!!

**Review by NickSNFK — APPROVED** [review]

**Pull request merged**